### PR TITLE
NEO-2118 Add renderInsetTable prop

### PR DESCRIPTION
**Pull request by yangAtSpoken**
[Link to updated components in Deploy Preview](https://deploy-preview-502--neo-react-library-storybook.netlify.app/?path=/story/components-table--custom-actions)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the Deploy Preview link at the top of this PR (or remove it if not applicable)
- [ ] updated the Figma referense link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [ ] tagged `@avaya-dux/dux-design` if any visual changes have occurred
- [x] tagged `@avaya-dux/dux-devs`

`@avaya-dux/dux-devs`:  Add renderInsetTableProp

- add a column to hold chevron > for expanding action

Screenshot 2024-08-20 at 1 35 08 PM

## Summary by CodeRabbit

- **New Features**
	- Enhanced `DraggableTableRow` and `StaticTableRow` components with conditional rendering for inset tables.
	- Improved functionality of the `Table` component to support expandable rows and dynamic inset table rendering.
	- Updated `TableHeader` to conditionally render an additional header cell based on inset table presence.

- **Bug Fixes**
	- Improved accessibility with new `aria-labels` and enhanced keyboard navigation in table components.

- **Style**
	- Added and refined styles for table headers and cells to improve layout and visual consistency.

- **Documentation**
	- Updated translation capabilities with a new label for expanding or collapsing table rows, enhancing user interaction.

### CODE REVIEW

File: src/components/Table/DraggableTableRow.tsx [hunk 1]
```diff
@@ -33,8 +36,11 @@ export const DraggableTableRow = <T extends Record<string, any>>({
 	const preparedRowProps = row.getRowProps();
 	logger.debug("row.isSelected", row.isSelected);
 
-	// + handle and checkbox columns
-	const cellCount = row.cells.length + 1 + (checkboxTd ? 1 : 0);
+	const { hasInsetTable, renderInsetTable } = useContext(FilterContext);
+
+	// count dynamic columns
+	const cellCount =
+		row.cells.length + 1 + (checkboxTd ? 1 : 0) + (hasInsetTable ? 1 : 0);
```
Comment (joe-s-avaya) [inline src/components/Table/DraggableTableRow.tsx:43]: so simple, so readable 👍

File: src/components/Table/Table.stories.tsx [hunk 2]
```diff
@@ -439,13 +439,15 @@ export const TableInTabs = () => (
...
 	const [multiple, setMultiple] = useState(false);
+	const [expandable, setExpandable] = useState(false);
 
 	return (
 		<Table
 			{...FilledFields}
 			draggableRows={checked}
 			selectableRows={multiple ? "multiple" : "none"}
 			caption="Custom Actions"
+			renderInsetTable={expandable ? () => "inset table" : undefined}
```
Comment (joe-s-avaya) [inline src/components/Table/Table.stories.tsx:450]: Does this only allow a single static input for all columns? I'd like to see a story that puts out unique information per row based on the row data. Such as sub-table, or a `<p>{name}</p>`, or something similar.

File: src/components/Table/TableComponents/TableHeader.tsx [hunk 3]
```diff
@@ -58,7 +59,11 @@ export const TableHeader: TableHeaderComponentType = ({
 			<tr>
 				{draggableRows && (
 					<th className="neo-table__dnd-th">
-						<div role="button" aria-label={translations.dragHandle}>
+						<div
+							role="button"
+							tabIndex={-1}
```
Comment (joe-s-avaya) [inline src/components/Table/TableComponents/TableHeader.tsx:64]: we don't want this to be keyboard accessible?

Comment (joe-s-avaya) [reply src/components/Table/TableComponents/TableHeader.tsx:64]: same question for line 85

File: src/components/Table/Table_shim.css [hunk 4]
```diff
@@ -129,3 +129,20 @@
...
+	width: 16px;
+	padding: 0;
+}
+
+.neo-table td button.td-icon--expand {
+	display: flex;
+	height: 14px;
+	width: 14px;
+	align-items: center;
+	color: var(--base-black-100);
```
Comment (joe-s-avaya) [inline src/components/Table/Table_shim.css:143]: `--base-black-100` isn't a color, as far as I can tell (I did a "find all" on the project). Did you mean `--neo-base-100`? Either way, don't forget to add the dark mode color (`prefers-color-scheme: dark`).

**Review by joe-s-avaya — APPROVED** [review]
I have a few questions and recommendations, but nothing that I want to block on. Plus, this appears to be a "part 1" so I'm approving, just make note of my comments.

File: src/components/Table/Table.stories.tsx (hunk 2 above)
Comment (yangAtSpoken) [reply src/components/Table/Table.stories.tsx:450]: Will add  in next PR.

File: src/components/Table/TableComponents/TableHeader.tsx (hunk 3 above)
Comment (yangAtSpoken) [reply src/components/Table/TableComponents/TableHeader.tsx:64]: No, these are empty header cells.

File: src/components/Table/Table_shim.css (hunk 4 above)
Comment (yangAtSpoken) [reply src/components/Table/Table_shim.css:143]: Good catch, will fix in next pr.

**Pull request merged**